### ISSUE TITLE
Send-email-confirmation - Auth.form.error.email.invalid

### ISSUE THREAD

**Issue by Zeb88** [strapi/strapi#4711]
**Information**

Node.js version: v12.12.0
npm version: 6.13.0
Strapi version: 3.0.0-beta.17.8
Database: postgres v  2.2.5 (59)
Operating system: macOS 10.15
Email provider: strapi-provider-email-sendgrid

**What is the current behavior?**
Can't send an email to a user with the link to confirm the user. Requests to the endpoint "http://localhost:1337/auth/send-email-confirmation" fail with the following error:

```
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": [
    {
      "messages": [
        {
          "id": "Auth.form.error.email.invalid"
        }
      ]
    }
  ]
}
```

**Steps to reproduce the problem**
Make a POST request to /auth/send-email-confirmation with the specified body in the documentation. 

**What is the expected behavior?**
The email should be sent with the confirmation link

**Suggested solutions**
Not sure what the issue is

**Relevant Strapi Doc**
[https://strapi.io/documentation/3.0.0-beta.x/plugins/users-permissions.html#email-validation
](https://strapi.io/documentation/3.0.0-beta.x/plugins/users-permissions.html#email-validation)

**Comment by Zeb88** [comment]
Update: This seems to work with Sendmail, but fails with other email providers.

**Comment by lauriejim** [comment]
Hello! I just tried with Sendgrid and it works well.
Your providers settings are correct?

**Comment by StringsOfArrays** [comment]
Hi, I got the same issue when using the register mutation on graphql... I am using default settings, no clue what is wrong.

**Comment by lauriejim** [comment]
So if you use the default provider, It's because off this https://github.com/strapi/strapi/issues/4737

**Comment by StringsOfArrays** [comment]
Very interesting, thank you. I do understand why this problem exists, but it makes me even more confused why the sendemail action in its seperated form worked. Since I suppose I should've encountered the same problem there.

**Comment by JuniYadi** [comment]
same issue when using Amazon SES in 18.2

Url : http://localhost:1337/auth/forgot-password

- Request
```
{
	"email": "xxxxx@gmail.com"
}
```

- Responses
```
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": [
    {
      "messages": [
        {
          "id": "Auth.form.error.email.invalid"
        }
      ]
    }
  ]
}
```

**Comment by lauriejim** [comment]
Hello! @JuniYadi !
For this provider I don't know.
Can you add a `console.log` to see the error of your node module.
https://github.com/strapi/strapi/blob/master/packages/strapi-provider-email-amazon-ses/lib/index.js#L66

**Comment by santicros** [comment]
Hello,

Hey @lauriejim I'm having the same problem with Mailgun. With the default sendmail it works but every email goes to SPAM. With Mailgun it does not send and says `Auth.form.error.email.invalid`.

Here I put my settings at the admin panel and with the `console.log`

`[2020-02-22T18:37:27.822Z] debug POST /auth/forgot-password (2702 ms) 200
[2020-02-22T18:37:37.420Z] debug PUT /email/settings/development (32 ms) 200
Error: { Error: getaddrinfo ENOTFOUND https://api.eu.mailgun.net/v3/profefreelance.com/messages https://api.eu.mailgun.net/v3/profefreelance.com/messages:443
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:56:26)
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'https://api.eu.mailgun.net/v3/profefreelance.com/messages',
  host: 'https://api.eu.mailgun.net/v3/profefreelance.com/messages',
  port: 443 }
[2020-02-22T18:37:47.137Z] debug POST /auth/forgot-password (51 ms) 400`

**Comment by selfagency** [comment]
Having the same issue as @JuniYadi but I figured it out: In the form field to configure the plugin, make sure you have an `https://` in front of the endpoint URI.

**Comment by selfagency** [comment]
It would be great, btw, if this could be configured with env vars and not with having to store my API credentials in the DB.

**Comment by derrickmehaffy** [comment]
> It would be great, btw, if this could be configured with env vars and not with having to store my API credentials in the DB.

That is currently in the works @selfagency

**Comment by gavmck** [comment]
@santicros I just had the same issue as you. 

I ended up opening up the strapi-provider-email-mailgun package code where you can see it [booting up mailgun](https://github.com/strapi/strapi/blob/master/packages/strapi-provider-email-mailgun/lib/index.js#L39) using another package: [mailgun-js](https://www.npmjs.com/package/mailgun-js)

The mailgun-js package was quite well documented and explained the expected values in the inputs in Strapi.

tl;dr Your host should just be `api.eu.mailgun.net`

Works now!

**Comment by lauriejim** [comment]
Thank you for this message! I'm closing this topic.
